commit cf6487d0d656994cc43851b8c2384741e220f7ac
Author: Christian Lamparter <chunkeey@googlemail.com>
Date:   Fri Sep 17 22:47:28 2010 +0200

    carl9170: abort tasklet during usb reset
    
    This patch prevents the tasklet code from
    interfering while the firmware is down for
    an unscheduled maintenance.
    
    Signed-off-by: Christian Lamparter <chunkeey@googlemail.com>
    Signed-off-by: John W. Linville <linville@tuxdriver.com>

--- a/drivers/net/wireless/ath/carl9170/usb.c
+++ b/drivers/net/wireless/ath/carl9170/usb.c
@@ -367,6 +367,9 @@ static void carl9170_usb_tasklet(unsigne
 {
 	struct ar9170 *ar = (struct ar9170 *) data;
 
+	if (!IS_INITIALIZED(ar))
+		return;
+
 	carl9170_usb_rx_work(ar);
 
 	/*
